packages/plugins/model-train/image-classification-tensorflow-model-train/src/index.ts

```diff
@@ -0,0 +1,96 @@
+import { ImageDataset, ModelTrainType, TfJsLayersModel, ModelTrainArgsType, ImageDataLoader } from '@pipcook/pipcook-core';
+
+const boa = require('@pipcook/boa');
+const { tuple } = boa.builtins();
+const tf = boa.import('tensorflow');
+const AUTOTUNE = tf.data.experimental.AUTOTUNE;
+
+function loadImage(path: string) {
+  let image = tf.io.read_file(path);
+  image = tf.image.decode_jpeg(image, boa.kwargs({
+    channels: 3
+  }));
+  return image;
+}
+
+async function createDataset(dataLoader: ImageDataLoader, labelMap: {
+  [key: string]: number;
+}) {
+  const imageNames: string[] = [];
+  const labels: number[] = [];
+  const count = await dataLoader.len();
+  for (let i = 0; i < count; i++) {
+    const currentData = await dataLoader.getItem(i);
+    imageNames.push(currentData.data);
+    labels.push(tf.one_hot(currentData.label.categoryId, Object.keys(labelMap).length));
+  }
+  const pathDs = tf.data.Dataset.from_tensor_slices(imageNames);
+  const imageDs = pathDs.map(loadImage, boa.kwargs({
+    num_parallel_calls: AUTOTUNE
+  }));
+  const labelDs = tf.data.Dataset.from_tensor_slices(labels);
+  const imageLabelDs = tf.data.Dataset.zip(tuple([ imageDs, labelDs ]));
+  return imageLabelDs;
+}
+
+/**
+ * this is plugin used to train tfjs model with pascal voc data format for image classification problem.
+ * @param data : train data
+ * @param model : model loaded before
+ * @param epochs : need to specify epochs
+ * @param batchSize : need to specify batch size
+ * @param optimizer : need to specify optimizer
+ */
+const ModelTrain: ModelTrainType = async (data: ImageDataset, model: TfJsLayersModel, args: ModelTrainArgsType): Promise<TfJsLayersModel> => {
+  try {
+    const {
+      epochs = 10,
+      batchSize = 16,
+      saveModel
+    } = args;
+
+    const { trainLoader, validationLoader, metadata } = data;
+
+    let trainDataSet = await createDataset(trainLoader, metadata.labelMap);
+    const count = await trainLoader.len();
+
+    trainDataSet = trainDataSet.repeat();
+    trainDataSet = trainDataSet.batch(batchSize);
+    trainDataSet = trainDataSet.prefetch(boa.kwargs({
+      buffer_size: AUTOTUNE
+    }));
+
+    const trainConfig: any = {
+      epochs: epochs,
+      steps_per_epoch: parseInt(String(count / batchSize))
+    };
+
+    if (validationLoader) {
+      console.log('create validation dataset');
```

Comment:
'console.log' is inappropriate:(

Reply:
what the reason for this. What should be used instead